main/src/cgeo/geocaching/EditWaypointActivity.java

```diff
@@ -468,11 +468,12 @@ public void updateCoordinates(final Geopoint gp) {
     @Override
     public void saveCalculatorState(final CalcState calcState) {
         this.calcState = calcState;
+        this.userNote.setText(calcState.notes);
     }
 
     @Override
     public CalcState fetchCalculatorState() {
-        return calcState;
+        return new CalcState(calcState, userNote.getText().toString());
```

Comment:
I'm concerned about this line.
Can we be confident that 'calcState' will always be defined whenever this is called?  Previously I think it was fine for 'fetchCalculatorState()' to return "null" but this change requires that 'calcState' be defined.

Reply:
Good point. I'll have a look.

Reply:
Changing this to:
```    @Override
    public CalcState fetchCalculatorState() {
        if (calcState != null) {
            calcState = new CalcState(calcState, userNote.getText().toString());
        }

        return calcState;
    }
```
Seems to work fine.

Reply:
I see a null check has been implemented to satisfy #6816.

Any reason we're not updating the 'calcState' variable though.

Reply:
This is done when coming back from the calculator in `saveCalculatorState()`, right?

Reply:
Yes, it will be updated when it comes time to save, but is their any reason were keeping the old one around at all?

To be honest I'm still a little hazy as to exactly how references work in Java (I'm a C++ developer by trade) but the fact that 'calcState' still refers to the old object means it can't be collected by the garbage collector until it is replaced, no?  I'm sure it doesn't matter, but once the new CalcState has been constructed we don't have any use for the old one do we?

Reply:
I think garbage collection is not a problem here, because latest when the activity gets finalized, the calcState gets finalized, too.
But I think I've introduced another bug here. I'll check later.

Reply:
Sure, garbage collection is a very minor issue, but I don't see the value in having two different versions of the CalcState object (one that is stored in 'calcState' and one that is returned by 'fetchCalculatorState()').  Should they not be the same?